specs/DesignDiscussions/InputValidation.md

```diff
@@ -0,0 +1,260 @@
+
+##### Submit (and Invoke) Actions
+Submit actions will default to `"inputs": "All"`. This means that all inputs on the card will be validated and sent with this submission. Card authors can adjust this default behavior by setting the `inputs` property to a different value as described above.
+
+##### Toggle and ShowCard Actions
+In many, or even most cases, it likely doesn't make sense to validate on show card or toggle actions if those actions are simply showing more details that don't need to appear on the main card. In some cases, however, either show card or toggle may be used to reveal the next step in a progression through the card experience. The wizard card above is a clear example of this. The card author may want to validate the first "page" before allowing the user to proceed to the second. Event the FoodOrder card can be viewed as an example of using show cards to progress through an interaction. If the upper portion of the card contained addition inputs, the card author may want to validate those fields before proceeding to ask additional questions about the users order. 
+
+Because by default Toggles and ShowCards don't necessarily need to validate inputs, these actions will default to `"inputs": "None"`. We will, however, support the ability to set a different value to enforce validation of some or all of the inputs before allowing the user to take one of these actions.
+
+> Question: Is it confusing to have an `inputs` property on a type that doesn't actually return those inputs (only validates them)? Would we want to add support for returning inputs on other action types? Is this an indication we need a different name?
+
+##### OpenUrl
+It seems unlikely that there are many scenarios to validate inputs on OpenUrl, so the default value for `inputs` will be false. For consistency, however, we will allow `inputs` to be set on OpenUrl actions.
+
+#### Disabling Actions
+
+Actions that depend on validation of the inputs should be greyed out until all related inputs have been validated. Because we are validating on focus lost, it would be possible for a user to navigate through a form using tab, resulting in all error messages and validation being shown. Clicking on a non-greyed out submit button at this time would not give any additional feedback, just silently fail to submit. To avoid this, we should grey out the submit button until such time as all inputs pass validation.
+
+## Custom Rendering Error Messages
+Comments in [#3081](https://github.com/microsoft/AdaptiveCards/issues/3081) from dclaux:
+> showInputValidationErrors: this needs to be introduced as well as a way for a host to custom render these error messages. We've had a very long debate around this.
+
+And andrewleader:
+>showInputValidationErrors for renderers (via code) makes sense
+
+TODO: Follow up on specifying this behavior.
```

Comment:
Discussion: This is about custom rendering the errors, for example in a pop up bubble rather then underneath. David feels this is important, but may be P2 for other renderers. Because this is a renderer feature, not a card schema thing, we'd be okay with an inconsistency between renderers if this is implemented for Javascript but not others.

Reply:
Put this context in the spec.

---
In reply to: [367155556](https://github.com/microsoft/AdaptiveCards/pull/3663#discussion_r367155556) [](ancestors = 367155556)